### PR TITLE
[Home Screen] Map/List Toggle Bar

### PR DESCRIPTION
**Pull request by jjiang17**
## Summary
Added toggle button to home screen that allows user to alternate between map view of the farm and list view of trees on the farm, with the map screen as the default view. When the user types in the search bar, the toggle automatically shifts to the list view to show the filtered results. 

## Test Plan
Try switching between map and list views in the home screen. Check that the search functionality and navigation to the appropriate TreeDetails screens work. 

## Notes
(You can leave this empty if you don't have anything.)

### Next Steps

Clean up code in HomeScreen, additional styling(?) 

### Relevant Links

Used this source for toggle functionality: https://github.com/mymai91/react-native-toggle-element 

## Screenshots and Tests

https://user-images.githubusercontent.com/71291124/156310674-04902685-03d0-4f6f-a2e9-f9bc704fcf11.MOV

### CODE REVIEW

**Review by daniellenwong — COMMENTED** [review]
Is it possible to have the buttons switch like this rather than sliding over to the next label? The sliding can get confusing but this might just be a nit because the user probably won't toggle between pages repeatedly.
https://www.figma.com/proto/Kgc8Qqw3eqtceSNyL9VZIg/Untitled?page-id=0%3A1&node-id=0%3A1&viewport=244%2C48%2C0.61&scaling=scale-down&starting-point-node-id=0%3A61

File: src/components/SearchCard.js [hunk 1]
```diff
@@ -1,18 +1,70 @@
...
+        flexDirection: 'row',
+        justifyContent: 'space-between',
+      }}
+      onPress={onPress}
+    >
+      <View>
+        <Text style={styles.title}>{name}</Text>
+        <Text style={styles.meta}>
+          {/* eslint-disable-next-line react/jsx-one-expression-per-line */}
+          {comments?.length || 0} comment
```
Comment (GuessingMan) [inline src/components/SearchCard.js:50]: maybe here we can use the Javascript f-string thing we did on akshay's PR

File: src/components/SearchCard.js [hunk 2]
```diff
@@ -1,18 +1,70 @@
...
+    </Card>
+  );
+}
 SearchCard.propTypes = {
-  name: PropTypes.string,
-  id: PropTypes.string,
-  onPress: PropTypes.func,
+  name: string,
+  // eslint-disable-next-line react/forbid-prop-types
+  comments: arrayOf(object),
```
Comment (GuessingMan) [inline src/components/SearchCard.js:68]: we can create more detailed props later

File: src/screens/ListScreen.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    height: '100%',
+    backgroundColor: '#eee',
+    paddingTop: 140,
+  },
+});
+
+export default function ListScreen({ navigation, style, data }) {
+  return (
+    <ScrollView style={[styles.list, style]}>
+      <View style={{ marginTop: 48, marginBottom: 248 }}>
```
Comment (GuessingMan) [inline src/screens/ListScreen.js:20]: this may become an issue if we have phones with different screen sizes, not 100% familiar with css padding however

**Review by GuessingMan — COMMENTED** [review]
overall looks good

1) noticed some css padding was hard coded and i'm not sure if that will be a big deal with different screen sizes.
2) we have to create more detailed proptypes, but that's for a later PR

**Review by GuessingMan — COMMENTED** [review]
looks good, we can address css and proptypes issues later

**Review by kbrgl — APPROVED** [review]

**Pull request merged**